h2o-automl/src/main/java/ai/h2o/automl/Leaderboard.java

```diff
@@ -430,18 +430,22 @@ Model getLeader() {
   }
 
   /**
-   * Delete everything in the DKV that this points to.  We currently need to be able to call this after deleteWithChildren().
+   * Delete object and its dependencies from DKV, including models.
    */
-  void delete() {
+  @Override
+  protected Futures remove_impl(Futures fs) {
+    for (Key<Model> m : models) {
+      Model model = m.get();
+      if (model != null) {
+        model.deleteCrossValidationPreds();
+        model.deleteCrossValidationModels();
+        model.deleteCrossValidationFoldAssignment();
+      }
```

Comment:
@michalkurka As you can see, when deleting a model, the CV details are not automatically deleted today.
I think you didn't want to include this to `model.delete()`: can you please remind me the reason for this?

Reply:
the argument was backwards compatibility, I don't think it is the best behavior but it could be a behavior people could be relying on

- imho we can transition to a better behavior - first give an option to delete with the CV models and later make that default

WDYT?

Reply:
@michalkurka this would be ideal transition, although we have to come up with something generic as RESTFul/client APIs are functional (as opposed to OO) so we have a limited way to extend it in a way that makes sense for every object.
For example in Python, we could use sth like:
```h2o.remove(model, cascade=False)```
as soon as it makes sense for all objects that can be deleted: frames, models, automl.
For frames, I guess the param could be ignored.
For models, `cascade=True` would additionally delete those CVs objects.
For AutoML, we could decide not to delete models and their dependencies if `cascade=False`.

I could try that: will submit a proposal.

Reply:
I very much like the `cascade=True` - IMHO that is a great way to go.